dfget/core/downloader/back_downloader/back_downloader.go

```diff
@@ -41,6 +40,8 @@ type BackDownloader struct {
 	Total   int64
 	Success bool
 
+	cfg *config.Config
```

Comment:
Does `BackDownloader` will use all fields in `config.Config`?

I do not think that it is a good way to use `config.Config` in `BackDownloader`. Since other fields in `BackDownloader`, like `URL`, `MD5` and so on are also in `config.Config`. Then it would be mixture when find a parameter which is both in `BackDownloader` and `BackDownloader.cfg`. I do not think it is a good habit to programme.

While for here, I think it is good to do like this:

* make clear that which parameters BackDownloader needs exactly;
* try to encapsulate all the parameter above into some field like `BackDownloaderConfig`;
* do a thoroughly initialization of  `BackDownloaderConfig` at the very beginning of `BackDownloader`.

I think data structure's purity is quite important for readiness of code. @lowzj

Reply:
I agree that data structure's purity is important for reading. Actually, the fields in `BackDownloaderConfig` are the public fields in `BackDownloader`, and they are initialized from `config.Config` in constructor `NewBackDownloader`.

Maybe we could open an issue and another pull request to do this.

Reply:
Yes, I will put forward an issue soon.